MetaMask/metamask-extension
# [FLASK] Add updated version of the Snaps settings UI

## Explanation

Fixes: https://github.com/MetaMask/snaps-monorepo/issues/1170

This PR will add redesigned UI for Snaps list and View Snap pages located withing MM Extension Settings.

Design specification: https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?node-id=18-6931&t=QWDDhu3X0w4bCUvC-0

## Screenshots/Screencaps
### Before
![Screenshot 2023-04-11 at 19 23 07](https://user-images.githubusercontent.com/13301024/231241603-78b36f54-9b17-4003-89ad-f49ab5c6fdc7.png)
![Screenshot 2023-04-11 at 19 24 00](https://user-images.githubusercontent.com/13301024/231241634-6176cabf-d9e3-4ba9-97bb-a9fdc53074dc.png)

### After
![Screenshot 2023-04-11 at 18 20 51](https://user-images.githubusercontent.com/13301024/231227215-4abc5703-8611-4acd-b271-fdd7e0840c0a.png)
![Screenshot 2023-04-11 at 19 12 02](https://user-images.githubusercontent.com/13301024/231239241-3a74259b-51e5-4367-b57e-27bc438d72b3.png)
![Screenshot 2023-04-11 at 19 13 53](https://user-images.githubusercontent.com/13301024/231239305-134b2ca3-a307-4302-b794-d7b2600d57a3.png)
![Screenshot 2023-04-11 at 19 14 22](https://user-images.githubusercontent.com/13301024/231239333-8c7c08f4-e440-4bc3-ada9-2d19786247e5.png)


## Manual Testing Steps
1. Go to Settings
2. Go to Snaps
3. Check the snaps list UI
4. Click on some of the listed snaps
5. Check the specific settings overview of the selected snap

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
